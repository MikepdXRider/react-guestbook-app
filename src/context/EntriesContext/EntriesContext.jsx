import React, { createContext, useContext, useState } from 'react'


const EntriesContext = createContext();

const EntriesProvider = ({children}) => {
    const [entries, setEntries] = useState([]);

    return(
        <EntriesContext.Provider value={{entries, setEntries}}>
            {children}
        </EntriesContext.Provider>
    )
}

const useEntries = () => {
    const context = useContext(EntriesContext);
    
    if (context === undefined) throw new Error('useEntries must be declared within an EntriesProvider');
    
    return context;
}

export { EntriesProvider, useEntries }