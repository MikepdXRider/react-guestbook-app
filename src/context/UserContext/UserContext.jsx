import React, { createContext, useContext, useState } from 'react'


const UserContext = createContext();

const UserProvider = ({children}) => {
    const [username, setUsername] = useState('');

    return(
        <UserContext.Provider value={{username, setUsername}}>
            {children}
        </UserContext.Provider>
    )
}


// 
// Could move to useAuth.jsx in the hooks folder. 
// 
const useUser = () => {
    const context = useContext(UserContext);

    if (context === undefined) throw new Error('useUser must be declared within an UserProvider');

    return context;
}

export { UserProvider, useUser }