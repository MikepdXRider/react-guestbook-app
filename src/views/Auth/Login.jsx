// 
// ✔ Login form here.
// ✔ Import useUser hook. 
// on login attempt, check to see if user input matches .env file info. If falsey, give error. If truthy, set user and redirect to the previous page or landing page.
// 

import React from 'react'
import { useUser } from '../../context/UserContext/UserContext.jsx'

export default function Login() {
    const {username, setUsername} = useUser();

    //  *Could be a replaced by a super-duper custom hook instead. Check out previous lab(react-treehouse-lab)
    // handleEmailChange
    // handlePasswordChange

    return (
        <fieldset className="w-1/4 border p-4">
            <legend>Sign In</legend>
            <form className="grid grid-cols-[1fr_2fr] grid-rows-3">
                <label htmlFor="username" className="self-center text-right">
                    Username
                </label>
                <input
                id="username"
                type="text"
                name="username"
                className="border p-2 m-3"
                required
                />

                <label htmlFor="password" className="self-center text-right">
                Password
                </label>
                <input
                id="password"
                type="password"
                name="password"
                className="border p-2 m-3"
                required
                />

                <button
                type="submit"
                className="h-12 self-center bg-black text-white p-1 font-bold col-span-2"
                >
                Sign In
                </button>
            </form>
            <p className="text-center text-red-500 font-bold">Error message goes here</p>
        </fieldset>
    )
}

