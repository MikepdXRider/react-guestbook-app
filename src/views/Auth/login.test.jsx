import react from "react";
import { render } from '@testing-library/react';
import { UserProvider } from '../../context/UserContext/UserContext.jsx';
import { EntriesProvider } from '../../context/EntriesContext/EntriesContext.jsx';
import Login from "./Login.jsx";

it('renders the login view', () => {
    const {container} = render(
            <UserProvider>
                <EntriesProvider>
                    <Login />
                </EntriesProvider>
            </UserProvider>
    );

    expect(container).toMatchSnapshot();
});