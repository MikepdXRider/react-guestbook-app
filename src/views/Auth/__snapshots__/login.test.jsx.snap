// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the login view 1`] = `
<div>
  <fieldset
    class="w-1/4 border p-4"
  >
    <legend>
      Sign In
    </legend>
    <form
      class="grid grid-cols-[1fr_2fr] grid-rows-3"
    >
      <label
        class="self-center text-right"
        for="password"
      >
        Username
      </label>
      <input
        class="border p-2 m-3"
        id="username"
        name="username"
        required=""
        type="text"
        value=""
      />
      <label
        class="self-center text-right"
        for="password"
      >
        Password
      </label>
      <input
        class="border p-2 m-3"
        id="password"
        name="password"
        required=""
        type="password"
        value=""
      />
      <button
        class="h-12 self-center bg-black text-white p-1 font-bold col-span-2"
        type="submit"
      >
        Sign In
      </button>
    </form>
  </fieldset>
</div>
`;
